packages/sdk/src/types/index.ts

```diff
@@ -24,9 +24,9 @@ export type CrossChainAction = {
   target: Address;
   callData: Hex;
   value: Amount;
-  update?: (outputAmount: bigint) => { callData: Hex; value: bigint };
-  updateCallData?: (outputAmount: bigint) => Hex;
-  updateValue?: (outputAmount: bigint) => bigint;
+  update?:
+    | ((outputAmount: bigint) => Promise<{ callData?: Hex; value?: bigint }>)
+    | ((outputAmount: bigint) => { callData?: Hex; value?: bigint });
```

Comment:
- can be async or not
- returns a partial, so you can update either `calldata` or `value` or both